src/OrleansRuntime/ReminderService/LocalReminderService.cs

```diff
@@ -202,17 +201,30 @@ public Task Stop()
         {
             if (stoppedCancellationTokenSource.IsCancellationRequested) return;
 
-            // try to retrieve reminder from all my subranges
-            myRange = ring.GetMyRange();
-            if (logger.IsVerbose2) logger.Verbose2("My range= {0}", myRange);
-            var acks = new List<Task>();
-            foreach (SingleRange range in RangeFactory.GetSubRanges(myRange))
+            using (await lockable.LockAsync())
             {
-                if (logger.IsVerbose2) logger.Verbose2("Reading rows for range {0}", range);
-                acks.Add(ReadTableAndStartTimers(range));
+                // try to retrieve reminder from all my subranges
+                var myRange = ring.GetMyRange();
```

Comment:
If we could compare `myRange` to its previous (saved) value, we could log here explicitly that the range has changed and differentiate those cases from periodic refreshes from the table. Not sure if it's overkill. Just a thought.